### PR TITLE
Use redux in react - add redux dispatch to react components

### PR DESCRIPTION
**Pull request by ram1117**
Hi, I have implemented the following in this PR. Please review the PR and provide feedback. Thank you.

- connect store to the react main component.
- Add books list to the redux state.
- Display books list in books component using redux books slice.
- Add add and remove books functionality to the application
- Add category input to the form. 
- Add error message after form validation.

Peek 2023-02-28 11-17

### CODE REVIEW

File: src/components/Categories.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
-  <button type="button" className="check-status">Check Status</button>
-);
+import { useSelector } from 'react-redux';
+
+const Categories = () => {
+  const status = useSelector((state) => state.category.status);
+  return (
+    <>
+      <h1>{status}</h1>
+      <button type="button" className="check-status">Check Status</button>
```
Comment (TedLivist) [inline src/components/Categories.js]: - [ ] This button on the categories page should dispatch an action when clicked. Right now the "Under construction" string is readily displayed when I visit the page. But that should not be so. The text should not be visible until I dispatch an action to the store which displays the text. Kindly refactor for that.

**Review by TedLivist — APPROVED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ram1117,

Good job so far!
✔️ Use of local state for form
✔️ Implementation of book removal and addition
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js (hunk 1 above)
Comment (ram1117) [reply src/components/Categories.js]: @TedLivist Hi Ted, thank you for the feedback. I will implement the review changes.

**Comment by ram1117** [comment]
Hi, I made the changes requested by previous code reviewer. Please review and provide feedback.

Peek 2023-02-28 13-03

**Review by OLIPLICHE — APPROVED** [review]
Hi @ram1117, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes requested by the previous review👏 👏
- Linter is green (the last commit have a green check mark (✅)) 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**